src/server/editorServices.ts

```diff
@@ -315,6 +315,10 @@ namespace ts.server {
             return this.host.directoryExists(path);
         }
 
+        getDirectories(path: string): string[] {
+            return this.host.getDirectories ? this.host.getDirectories(path) : [];

```

Comment:
why would the host not have this function? it is on sys, and we are the only implementers of this API so far.